tests/integration/new_relations/test_new_relations.py

```diff
@@ -362,9 +362,16 @@ async def test_relation_data_is_updated_correctly_when_scaling(ops_test: OpsTest
         )
 
         # Remove the original units.
-        await ops_test.model.applications[DATABASE_APP_NAME].destroy_units(*units_to_remove)
+        leader_unit = await get_leader_unit(ops_test, DATABASE_APP_NAME)
+        await ops_test.model.applications[DATABASE_APP_NAME].destroy_units(*[
+            unit for unit in units_to_remove if unit != leader_unit.name
+        ])
         await ops_test.model.wait_for_idle(
-            apps=[DATABASE_APP_NAME], status="active", timeout=1500, wait_for_exact_units=2
+            apps=[DATABASE_APP_NAME], status="active", timeout=600, wait_for_exact_units=3
+        )
+        await ops_test.model.applications[DATABASE_APP_NAME].destroy_units(leader_unit.name)
+        await ops_test.model.wait_for_idle(
+            apps=[DATABASE_APP_NAME], status="active", timeout=600, wait_for_exact_units=2
         )
```

Comment:
Destroy the original follower unit and then the leader

Reply:
That would be a good addition to ensure that the connection details in the relation databag are updated correctly, even if the primary is not the Juju leader.

Another thing to remember: add a note to https://warthogs.atlassian.net/browse/DPE-3684 to add the test back that deletes the leader and the synchronous standby (so we can ensure that if that happens - two units at the same time - the cluster doesn't get stuck).

Reply:
I would say those extensive tests are absolutely valid and diverse the ticket in backlog to be implemented. At the moment we are trying to stabilize CI test to move forward.

Reply:
Added a comment to the ticket.